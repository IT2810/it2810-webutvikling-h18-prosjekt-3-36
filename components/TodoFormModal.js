import React from 'react';
import {
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  Alert,
  Button,
  TextInput,
  SafeAreaView
} from 'react-native';
import Modal from "react-native-modal";
import DatePicker from 'react-native-datepicker';
import { FloatingAction } from 'react-native-floating-action';

export default class TodoForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {isModalVisible: false}
  }

  _toggleModal = () => {
    this.setState({ text: null });
    this.setState({ date: null });
    // clear the text and date aswell as to toggle whether the modal is visible or not.
    this.setState({ isModalVisible: !this.state.isModalVisible });
  }

  addNewTodo = () => {
    // Kjør addNewTodo metoden til TodoScreen så alt av todo states blir gjort av den.
    this.props.addNewTodo(this.state.text, this.state.date, this.state.categoryChosen);
    this._toggleModal();
  }

  render() {
    // actions er hvilken handlinger du kan gjøre etter du har trykker på den store + knappen. Vi har valgt å ha kategorier her. 
    const actions = [{
      text: 'Work',
      name: 'work',
      color: "red",
      position: 1
    }, {
      text: 'School',
      name: 'school',
      color: "blue",
      position: 2
    }, {
      text: 'Fun',
      name: 'fun',
      color: "orange",
      position: 3
    }, {
      text: 'Other',
      name: 'other',
      color: "green",
      position: 4
    }];
  return (
    <SafeAreaView style={styles.container}>
      <Modal isVisible={this.state.isModalVisible}>
        <View style={{ flex: 1, backgroundColor: "white" }}>
          <View style={styles.buttonContainer}>
            <View style={styles.button} ><Button title="Cancel" onPress={this._toggleModal} color="red" style={styles.button}></Button></View>
            <View style={styles.button} ><Button title="Submit" onPress={this.addNewTodo} color="green" style={styles.button}></Button></View>
          </View>
          <TextInput
            editable={true}
            multiline
            onChangeText={(text) => this.setState({ text })}
            value={this.state.text}
            placeholder="Write something..."
            placeholderTextColor="grey"
            style={styles.textInput}
            underlineColorAndroid='rgba(0,0,0,0)'
            onSubmitEditing={this.addNewTodo}
            autoFocus={true}></TextInput>
          <DatePicker
            style={{ width: 200 }}
            date={this.state.date}
            mode="date"
            placeholder="select date"
            format="YYYY-MM-DD"
            minDate="1990-01-01"
            maxDate="2025-01-01"
            confirmBtnText="Confirm"
            cancelBtnText="Cancel"
            customStyles={{
              dateIcon: {
                position: 'absolute',
                left: 0,
                top: 4,
                marginLeft: 0
              },
              dateInput: {
                marginLeft: 36
              }
            }}
            onDateChange={(date) => { this.setState({ date: date }) }}
          />
          <Text>Add a date for your todo (optional)</Text>
        </View>
      </Modal>
      <View style={styles.container}>
        <FloatingAction
          actions={actions}
          position="right"
          onPressItem={
            (name) => {
              this.setState({ isModalVisible: true });
              this.setState({ categoryChosen: name });
            }
          }
        />
      </View>
    </SafeAreaView>
  );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10,
  },
  button: {
    width: '45%',
  },
  textInput: {
    fontSize: 18,
  }
});
