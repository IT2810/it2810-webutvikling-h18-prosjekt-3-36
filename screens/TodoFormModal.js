import React from "react";
import { StyleSheet, Text, View, Alert, Button, TextInput, Platform } from "react-native";
import HeaderButton from "../components/HeaderButton";
import DatePicker from "react-native-datepicker";
import { Modal } from "react-native";

export default class TodoForm extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      headerTitle: "test",
      headerLeft: <HeaderButton onPress={() => navigation.goBack()} title="Cancel" />,
      headerRight: <HeaderButton onPress={() => navigation.add()} disabled={true} title="Add" />
    };
  };

  constructor(props) {
    super(props);
    this.state = {
      helpAlert: true
    };
  }

  componentWillMount() {
    this.props.navigation.setParams({ add: this.add });
  }

  add = () => {
    console.log("save");
  };

  addNewTodo = () => {
    // Kjør addNewTodo metoden til TodoScreen så alt av todo states blir gjort av den.
    if (this.state.helpAlert && this.state.date === undefined) {
      // Om brukern ikke har valgt en dato burde det komme en liten varsling om at den ikke kan bli sett i kalendern.
      Alert.alert(
        "No date chosen",
        "You have not chosen a date and will therefore not be able to view your todo in Calender mode, swipe right to view your todos without date.",
        [{ text: "Never remind me again", onPress: () => this.setState({ helpAlert: false }) }, { text: "OK" }],
        { cancelable: false }
      );
    }
    this.props.addNewTodo(this.state.text, this.state.date);
    this.props.toggleModal();
    this.clearForm();
  };

  clearForm = () => {
    this.setState({ text: undefined });
    this.setState({ date: undefined });
  };

  render() {
    return (
      <View style={{ flex: 1, backgroundColor: "white" }}>
        <TextInput
          editable={true}
          multiline
          style={{ minHeight: 200 }}
          onChangeText={text => this.setState({ text })}
          value={this.state.text}
          placeholder="Write something..."
          placeholderTextColor="grey"
          autoFocus={true}
          underlineColorAndroid="rgba(0,0,0,0)"
          onSubmitEditing={this.addNewTodo}
        />
        <DatePicker
          style={{ width: 200 }}
          date={this.state.date}
          mode="date"
          placeholder="select date"
          format="YYYY-MM-DD"
          minDate="1990-01-01"
          maxDate="2025-01-01"
          confirmBtnText="Confirm"
          cancelBtnText="Cancel"
          customStyles={{
            dateIcon: {
              position: "absolute",
              left: 0,
              top: 4,
              marginLeft: 0
            },
            dateInput: {
              marginLeft: 36
            }
          }}
          onDateChange={date => {
            this.setState({ date: date });
          }}
        />
        <Text>Add a date for your todo (optional)</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  buttonContainer: {
    flex: 1,
    maxHeight: 50,
    flexDirection: "row",
    justifyContent: "space-around",
    padding: 10
  },
  button: {
    width: "45%"
  },
  textInput: {
    fontSize: 18
  }
});
