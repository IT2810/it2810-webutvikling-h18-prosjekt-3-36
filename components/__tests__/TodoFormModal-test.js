import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import TodoFormModal from "../TodoFormModal";


it("TodoFormModal Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TodoFormModal />)
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
});
