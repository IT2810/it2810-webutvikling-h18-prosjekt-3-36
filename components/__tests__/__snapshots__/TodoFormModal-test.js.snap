// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFormModal Matches snapshot. 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <Component
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 10,
        }
      }
    >
      <Component
        style={
          Object {
            "width": "45%",
          }
        }
      >
        <Button
          color="red"
          style={
            Object {
              "width": "45%",
            }
          }
          title="Cancel"
        />
      </Component>
      <Component
        style={
          Object {
            "width": "45%",
          }
        }
      >
        <Button
          color="green"
          onPress={[Function]}
          style={
            Object {
              "width": "45%",
            }
          }
          title="Submit"
        />
      </Component>
    </Component>
    <TextInput
      allowFontScaling={true}
      autoFocus={true}
      editable={true}
      multiline={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Write something..."
      placeholderTextColor="grey"
      style={
        Object {
          "fontSize": 18,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
    />
    <DatePicker
      TouchableComponent={[Function]}
      allowFontScaling={true}
      androidMode="default"
      cancelBtnText="Cancel"
      confirmBtnText="Confirm"
      customStyles={
        Object {
          "dateIcon": Object {
            "left": 0,
            "marginLeft": 0,
            "position": "absolute",
            "top": 4,
          },
          "dateInput": Object {
            "marginLeft": 36,
          },
        }
      }
      date=""
      disabled={false}
      duration={300}
      format="YYYY-MM-DD"
      height={259}
      hideText={false}
      iconSource={1}
      maxDate="2025-01-01"
      minDate="1990-01-01"
      modalOnResponderTerminationRequest={[Function]}
      mode="date"
      onDateChange={[Function]}
      placeholder="select date"
      showIcon={true}
      style={
        Object {
          "width": 200,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Add a date for your todo (optional)
    </Text>
  </Component>
</ReactNativeModal>
`;
